inst/inst_CRIRES.py

```diff
@@ -109,6 +111,75 @@ def Tell(molec):
       return w_atm, f_atm
 
 
+def write_fits(wtpl_all, tpl_all, e_all, list_files, file_out):
+
+    file_in = list_files[0]
+
+    # copy header from first fits file 
+    hdu = fits.open(file_in, ignore_blank=True)
+    hdr = hdu[0].header
+
+    if len(list_files) > 1:
+        # delete parts that vary for all observations
+        del hdr['DATE-OBS']
+        del hdr['UTC']
+        del hdr['LST']
+        del hdr['ARCFILE']
+        del hdr['ESO INS SENS*']
+        del hdr['ESO INS TEMP*']
+        del hdr['ESO INS1*']
+        del hdr['ESO DET*']
+        del hdr['ESO OBS*']
+        del hdr['ESO TPL*']
+        del hdr['ESO TEL*']
+        del hdr['ESO OCS MTRLGY*']
+        del hdr['ESO ADA*']
+        del hdr['ESO AOS*']
+        del hdr['ESO SEQ*']
+        del hdr['ESO PRO DATANCOM']
+        del hdr['ESO PRO REC1 PARAM*']
+        del hdr['ESO PRO REC1 RAW*']	
+
+        for hdri in hdu:
+            hdri.header['EXPTIME'] = 0
+
+    # file creation date 
+    now = datetime.now()
+    dt_string = now.strftime("%Y-%m-%dT%H:%M:%S")
+    hdr['DATE'] = dt_string
+
+    # save raw file informations in FITS header
+    hdr.set('ESO PRO REC2 ID', 'viper_create_tpl', 'Pipeline recipe', after='ESO PRO REC1 PIPE ID')
+
+    for i in range(0, len(list_files), 1):
+        pathi, filei = os.path.split(list_files[len(list_files)-i-1])
+        hdr.set('ESO PRO REC2 RAW'+str(len(list_files)-i)+' NAME', filei, 'File name', after='ESO PRO REC2 ID')
+
+    hdr.set('ESO PRO DATANCOM', len(list_files), 'Number of combined frames', after='ESO PRO REC2 RAW'+str(len(list_files))+' NAME')
+
+    # write the template data to the file
+    for o in range(1,19,1): 
+        # data spread over 3 detectors, each having 6 orders
+        oi = 5-int((o-1)/3)  
+        d = np.mod(o,3) 
```

Comment:
+ `# detector number`

Reply:
So what is `d` if it not the detector number?

Reply:
Sorry, I think I misunderstood. d is indeed the detector number and if you just wanted to add this comment here, it would be a good idea.